services/dynamic/dynamic-yaml.service.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+const yaml = require('js-yaml')
+const jp = require('jsonpath')
+const emojic = require('emojic')
+const BaseService = require('../base')
+const { InvalidResponse } = require('../errors')
+const trace = require('../trace')
+const {
+  createRoute,
+  queryParamSchema,
+  errorMessages,
+  renderDynamicBadge,
+} = require('./dynamic-helpers')
+
+module.exports = class DynamicYaml extends BaseService {
+  static get category() {
+    return 'dynamic'
+  }
+
+  static get route() {
+    return createRoute('yaml')
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'custom badge',
+    }
+  }
+
+  parseYml(buffer) {
```

Comment:
I think it would be useful to extract this as a generic helper function so we can use it in the F-Droid service (and in a `BaseYamlService` in future)

Its a bit scope creep-ey though, so we could merge this PR and raise it as an issue for future so that we don't have to fiddle with the F-Droid badge in this PR, if you want.

Reply:
Makes sense. That came up in https://github.com/badges/shields/pull/2399#issuecomment-444970457, as well.